installers/rpm-code/codes/mantid.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+mantid_main() {
+    codes_yum_dependencies \
+        eigen3-devel \
+        OCE-devel \
+        jemalloc-devel \
+        jsoncpp-devel \
+        muParser-devel \
+        poco-devel
+    codes_dependencies common boost ipykernel
+    install_pip_install pre-commit pylint sphinx_bootstrap_theme
+    mantid_nexus_install
+    # Mantid installs a great deal and requires its own specific environment variables to run right
+    # (ex. see the mantidpython script it creates). So, install it in its own subdirectory so it
+    # doesn't interfere with other libs.
+    local p="${codes_dir[lib]}/mantid"
+    mantid_install "$p"
+    mantid_ipykernel $(mantid_install_rsmantid "$p")
+}
+
+mantid_install() {
+    local install_prefix="$1"
+    codes_download mantidproject/mantid
+    mantid_patch_eigen_cmake
+    codes_cmake \
+        -DBOOST_ROOT="${codes_dir[prefix]}" \
+        -DCMAKE_INSTALL_PREFIX="$install_prefix" \
+        -DENABLE_WORKBENCH=OFF \
+        -DMANTID_FRAMEWORK_LIB=BUILD \
+        -DMANTID_QT_LIB=OFF \
+        -DNEXUS_CPP_LIBRARIES="${codes_dir[lib]}/libNeXusCPP.so" \
```

Comment:
This is always interesting. The only thing needing quoting is between the braces. Some people there is some special syntax with `--key=value` args, but the shell knows nothing about this (caveat for interactive autocomplete) so I would say as an "example" it's better to quote just what's needed when there's only one thing quoted. Obviously with multiple substitutions, then quote the whole arg (including --key).

Reply:
8c3e1a0 thanks for the details. I wasn't sure what our preferred style was...